content/academy/web_scraping_for_beginners/best_practices.md

```diff
@@ -0,0 +1,126 @@
+---
+
+# [](#best-practices) Best practices when writing scrapers
+
+Every developer has their own style, which evolves as they grow and learn. While one dev might prefer a more  [functional](https://en.wikipedia.org/wiki/Functional_programming) style, another might find an [imperative](https://en.wikipedia.org/wiki/Imperative_programming) approach to be more intuitive. We at Apify understand this, and have written this best practices lesson with that in mind.
+
+The goal of this lesson is not to force you into a specific paradigm or to make you think that you're doing things wrong, but instead to provide you some insight into the standards and best practices that we at Apify follow to ensure readable, maintainable, scalable code.
+
+## [](#code-style) Code style
+
+There are some general things we recommend when it comes to your code style when writing scrapers.
+
+### [](#clean-code) Clean code
+
+Praise [clean code](https://blog.risingstack.com/javascript-clean-coding-best-practices-node-js-at-scale/)! Use proper variable and function names that are descriptive of what they are, and split your code into smaller [pure](https://en.wikipedia.org/wiki/Pure_function) functions.
+
+### [](#constants) Constant variables
+
+Define any [constant variables](https://softwareengineering.stackexchange.com/questions/250619/best-practices-reasons-for-string-constants-in-javascript) that globally apply to the scraper in a single file named **constants.js**, from where they will all be imported. Constant variable names should be in `UPPERCASE_WITH_UNDERSCORES` style.
+
+> If you have a whole lot of constant variables, they can be in a folder named **constant** organized into different files.
+
+### [](#use-es6) Use modern ES6 JavaScript
+
+If you're writing your scraper in JavaScript, use [ES6](https://www.w3schools.com/js/js_es6.asp) features and ditch the old ones which they replace. This means using `const` and `let` instead of `var`, `includes` instead of `indexOf`, etc.
+
+> To learn more about some of the most popular (and awesome) ES6+ features, check out [this](https://medium.com/@matthiasvstephens/why-is-es6-so-awesome-88bff6857849) article.
+
+### [](#no-magic-numbers) No magic numbers
+
+Avoid using [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) as much as possible. Either declare them as a **constant** variable in your **constants.js** file, or if they are only used once, add a comment explaining what the number is.
+
+Don't write code lke this:
+
+```JavaScript
+const x = (y) => (y - 32) * (5 / 9);
+```
+
+That is quite confusing due to the undescriptive naming and the magic numbers. Do this instead:
+
+```JavaScript
+// Converts a fahrenheit value to celsius
+const fahrenheitToCelsius = (celsius) => (celsius - 32) * (5 / 9);
+```
+
+> Additionally, try to make values for options such as `maxRequestRetries` in the Apify SDK configurable via input. If they absolutely must be a strict value, make them a constant variable.
```

Comment:
I think this is a bit out of place here in the beginners tutorial. I would just get rid of it, I don't think it's super important.

Reply:
I would also skip it, I think right now the best practice is to not put as much stuff into input.